src/Telemetry/AzureDevOpsTimeline/AzureDevOpsTimeline.cs

```diff
@@ -231,6 +237,45 @@ private async Task RunLoop(CancellationToken cancellationToken)
                         issue.AugmentedIndex = "999." + issue.Index.ToString("D3");
                     }
                 }
+            }      
+
+            TimeSpan cancelationTime; 
+            if(!TimeSpan.TryParseExact(_options.Value.LogScrapingTimeout, @"m\:s\:fff", CultureInfo.CurrentCulture, out cancelationTime)){
+                cancelationTime = TimeSpan.FromMinutes(10);
+            }
+
+            try
+            {
+                Stopwatch stopWatch = new Stopwatch();
+
+                _logger.LogInformation("Starting log scraping");
+
+                var logScrapingTimeoutCancellationTokenSource = new CancellationTokenSource(cancelationTime);
+                var logScrapingTimeoutCancellationToken = logScrapingTimeoutCancellationTokenSource.Token;
+
+                var combinedCancellationTokenSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken, logScrapingTimeoutCancellationToken);
+                var combinedCancellationToken = combinedCancellationTokenSource.Token;
+
+                stopWatch.Start();
+
+                var task = Task.Run(async () =>
```

Comment:
There shouldn't be any need to use Task.Run here. And I'm not sure it does what is expected.  Just sending the cancellationToken with the timeout set should be sufficient.